### PR TITLE
feat(bounty): added view for hackathon, meme and webinar

### LINKED ISSUE #3: Viewing Individual Proposals and Bounties

### PR DESCRIPTION
**Pull request by pratham15**

### CODE REVIEW

File: components/dashboard/bounties/meme-completion-item.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          </Tr>
+        </Thead>
+        <Tbody>
+          {MemeCompletion &&
+            MemeCompletion.winners_info.map((submission, index) => (
+              <Tr key={index}>
+                <Td>{submission.name}</Td>
+                <Td>{submission.account_id}</Td>
+                <Td>
+                  <Link href={`${submission.submission_link}`} isExternal>
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/meme-completion-item.tsx]:
```tsx
<Link href={submission.submission_link} isExternal>
```

File: components/dashboard/bounties/meme-completion-item.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    Link{' '}
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/meme-completion-item.tsx]: Why do we have this empty string? If you need spacing use padding or margin.

File: components/dashboard/bounties/hackathon-completion-item.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  TableCaption,
+  Link,
+  Heading,
+  Text,
+  Flex,
+} from '@chakra-ui/react';
+import { HackathonCompletionBounty } from '../../../types';
+import { ItemDetailContainer } from '../item-detail-container';
+
+export type HackathonCompletionItemProp = {
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/hackathon-completion-item.tsx]: `HackathonCompletionItemProps`

File: components/dashboard/bounties/webinar-completion-item.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <>
+      <ItemDetailContainer
+        text="Number of Registrations"
+        value={WebinarCompletion.num_of_registrations}
+      />
+      <ItemDetailContainer
+        text="Number of Attendees"
+        value={WebinarCompletion.num_of_attendees}
+      />
+      <Link pt={2} isExternal href={`${WebinarCompletion.webinar_link}`}>
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/webinar-completion-item.tsx]:
```tsx
<Link pt={2} isExternal href={item.webinar_link}>
```

File: components/dashboard/bounties/webinar-completion-item.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export const WebinarCompletionItem: React.FC<WebinarCompletionItemProps> = ({
+  item: WebinarCompletion,
+}) => {
+  return (
+    <>
+      <ItemDetailContainer
+        text="Number of Registrations"
+        value={WebinarCompletion.num_of_registrations}
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/webinar-completion-item.tsx]: It is `item`

File: components/dashboard/bounties/webinar-completion-item.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      />
+      <ItemDetailContainer
+        text="Number of Attendees"
+        value={WebinarCompletion.num_of_attendees}
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/webinar-completion-item.tsx]: it is `item`

File: components/dashboard/bounties/hackathon-completion-item.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          </Tr>
+        </Thead>
+        <Tbody>
+          {HackathonCompletion &&
+            HackathonCompletion.winners_info.map((submission, index) => (
+              <Tr key={index}>
+                <Td>{submission.name}</Td>
+                <Td>{submission.account_id}</Td>
+                <Td>
+                  <Link href={`${submission.submission_link}`} isExternal>
```
Comment (siddharthborderwala) [inline components/dashboard/bounties/hackathon-completion-item.tsx]:
```tsx
<Link href={submission.submission_link} isExternal>
```

File: components/dashboard/proposals/hackathon-proposal-item.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { Box, Link } from '@chakra-ui/react';
+import { HackathonProposal } from '../../../types';
+import { ItemDetailContainer } from '../item-detail-container';
+
+export type HackathonProposalItemProps = {
+  item: HackathonProposal;
+};
+export const HackathonProposalItem: React.FC<HackathonProposalItemProps> = ({
+  item: HackathonProposal,
```
Comment (siddharthborderwala) [inline components/dashboard/proposals/hackathon-proposal-item.tsx:9]: Why have you renamed it? Just keep it as item, renaming has to benefit here, it is only gonna cost some performance.

I saw renames in other components as well, remove it from there as well.

File: components/dashboard/proposals/hackathon-proposal-item.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      />
+      <ItemDetailContainer
+        text="Expected Registrations"
+        value={HackathonProposal.expected_registrations}
+      />
+      <Box mt={2}>
+        <Link
+          color="blue.600"
+          isExternal
+          href={`${HackathonProposal.supporting_document}`}
```
Comment (siddharthborderwala) [inline components/dashboard/proposals/hackathon-proposal-item.tsx:25]:
```tsx
href={HackathonProposal.supporting_document}
```

File: pages/dashboard/proposals/[id].tsx [hunk 10]
```diff
@@ -14,10 +14,20 @@ import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { Plus } from 'phosphor-react';
 import React, { useEffect, useState } from 'react';
+import { HackathonCompletionItem } from '../../../components/dashboard/bounties/hackathon-completion-item';
+import { HackathonProposalItem } from '../../../components/dashboard/proposals/hackathon-proposal-item';
+import { MemeContestProposalItem } from '../../../components/dashboard/proposals/meme-contest-proposal-item';
+import { OpenProposalItem } from '../../../components/dashboard/proposals/open-proposal-item';
```
Comment (siddharthborderwala) [inline pages/dashboard/proposals/[id].tsx:19]:
```tsx
import { HackathonProposalItem, MemeContestProposalItem, OpenProposalItem } from '../../../components/dashboard/bounties/hackathon-completion-item';
```

Make all the named imports from the same file in the same import statement.

File: pages/dashboard/proposals/[id].tsx [hunk 11]
```diff
@@ -14,10 +14,20 @@ import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { Plus } from 'phosphor-react';
 import React, { useEffect, useState } from 'react';
+import { HackathonCompletionItem } from '../../../components/dashboard/bounties/hackathon-completion-item';
```
Comment (siddharthborderwala) [inline pages/dashboard/proposals/[id].tsx]: Why is bounty item imported here?

File: pages/dashboard/proposals/[id].tsx [hunk 12]
```diff
@@ -34,8 +44,8 @@ const ProposalItem: NextPage = () => {
     };
   }, [contract, id, setProposal]);
 
+  console.log(proposal);
```
Comment (siddharthborderwala) [inline pages/dashboard/proposals/[id].tsx:46]: remove un-used logs

File: types/contract.ts [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+export type ProposalType =
+  | {
+      [TypesOfProposals.HACKATHON]: HackathonProposal;
+    }
+  | {
+      [TypesOfProposals.MEME_CONTEST]: MemeContestProposal;
+    }
+  | {
+      Open: OpenProposal;
```
Comment (siddharthborderwala) [inline types/contract.ts]:
```tsx
{
  [TypesOfProposals.OPEN]: OpenProposal;
}
```

**Review by siddharthborderwala — COMMENTED** [review]
Make the changes I have commented.

File: components/dashboard/bounties/meme-completion-item.tsx (hunk 2 above)
Comment (pratham15) [reply components/dashboard/bounties/meme-completion-item.tsx]: I believe it was the prettier formatting that adds {' '} for some reason

File: components/dashboard/proposals/hackathon-proposal-item.tsx (hunk 8 above)
Comment (pratham15) [reply components/dashboard/proposals/hackathon-proposal-item.tsx:9]: Noted, I've renamed item to its original self everywhere

File: pages/dashboard/proposals/[id].tsx (hunk 10 above)
Comment (pratham15) [reply pages/dashboard/proposals/[id].tsx:19]: Exported all the components from `completion-items.ts`

**Review by siddharthborderwala — APPROVED** [review]

**Pull request merged**